cclib/parser/nboparser.py

```diff
@@ -136,8 +151,11 @@ def extract(self, inputfile, line):
                 total          = float(population_analysis[6])
                                 
                 # TODO append to attibutes
-                #             
-                # self.append_attribute('naos', nao)
+
+                if 'nbo' not in self.atomcharges:
+                    self.atomcharges["nbo"] = []

```

Comment:
I would put a variable like `charges = []` outside the loop and then have a single `self.atomcharges["nbo"] = charges` after the loop is done.